### PR TITLE
chore: Expose widget configuration to define properties that can set …

### LINKED ISSUE #16537: [Bug]: Table widget should not load while query bound to its column property is running
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Currently, the table widget goes into a loading state if a query bound to its column property is running. Ideally, the table widget should only load while the query bound to its tabledata property runs. 
A similar use case could appear in other widgets as well. Hence it would be good to let the widget define the loading properties instead of considering all the properties to be loading properties.

### Steps To Reproduce
https://www.loom.com/share/2e51ad710a42492a902c3db400795cfc
https://appsmith-27rdtyjqx-get-appsmith.vercel.app/app
1. Drag and drop a table widget
2. Add a select column
3.  Bind the options property of the column to a query
4. enable serverside searching.
5. search a value in the select. observe that the entire row gets loaded.

### Public Sample App

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by sbalaji1192**
…widget to loading state

## Description

Partially reinstates changes from the reverted commit - https://github.com/appsmithorg/appsmith/pull/11875/files

Fixes #16537

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sbalaji1192** [comment]
CC @ohansFavour @eco-monk

File: app/client/src/utils/WidgetLoadingStateUtils.test.ts [hunk 1]
```diff
@@ -72,14 +73,61 @@ const Query_tree: DataTreeAction = {
...
+  actionId: "",
+  config: {},
+  pluginType: PluginType.API,
+  pluginId: "",
+  name: "",
+  run: {},
+  clear: {},
+  dynamicBindingPathList: [],
+  bindingPaths: {},
+  ENTITY_TYPE: ENTITY_TYPE.ACTION,
```
Comment (Rishabh-Rathod) [inline app/client/src/utils/WidgetLoadingStateUtils.test.ts:87]: Can we add `reactivePaths` here, currently build is failing due to the type error.

Comment (sbalaji1192) [reply app/client/src/utils/WidgetLoadingStateUtils.test.ts:87]: Pushed the changes

File: app/client/src/utils/WidgetLoadingStateUtils.ts [hunk 2]
```diff
@@ -97,43 +98,57 @@ export const getEntityDependants = (
...
   // check animateLoading is active on current widgets and set
   const filteredLoadingEntityNames = new Set<string>();
-  loadingEntitiesDetails.names.forEach((entityName) => {
+
+  loadingEntityPaths.forEach((entityPath) => {
+    const entityPathArray = entityPath.split(".");
+    const entityName = entityPathArray[0];
+    const widgetType = get(dataTree, [entityName, "type"]);
+    const loadingProperties = WidgetFactory.getLoadingProperties(widgetType);
+
```
Comment (Rishabh-Rathod) [inline app/client/src/utils/WidgetLoadingStateUtils.ts:141]: Can we vefiry if the entity is of type widget before trying to get `loadingProperties` ?

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM. 

QQ: Do we only want this in TableV2 or we could add this to TableV1 as well?

**Comment by sbalaji1192** [comment]
We don't need it in Table V1. This issue is caused by the select column type which is not available in table v1.

**Review by Rishabh-Rathod — DISMISSED** [review]
LGTM

**Review by Rishabh-Rathod — APPROVED** [review]

**Comment by laveena-en** [comment]
Will be verifying this once select type column is up on release.

**Pull request merged**